src/Symfony/Component/Translation/Bridge/Loco/Provider/LocoProvider.php

```diff
@@ -0,0 +1,245 @@
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Bridge\Loco\Provider;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Translation\Exception\ProviderException;
+use Symfony\Component\Translation\Loader\LoaderInterface;
+use Symfony\Component\Translation\Provider\ProviderInterface;
+use Symfony\Component\Translation\TranslatorBag;
+use Symfony\Component\Translation\TranslatorBagInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Mathieu Santostefano <msantostefano@protonmail.com>
+ *
+ * In Loco:
+ *  * Tags refers to Symfony's translation domains
+ *  * Assets refers to Symfony's translation keys
+ *  * Translations refers to Symfony's translated messages
+ *
+ * @experimental in 5.3
+ */
+final class LocoProvider implements ProviderInterface
+{
+    private $client;
+    private $loader;
+    private $logger;
+    private $defaultLocale;
+    private $endpoint;
+
+    public function __construct(HttpClientInterface $client, LoaderInterface $loader, LoggerInterface $logger, string $defaultLocale, string $endpoint)
+    {
+        $this->client = $client;
+        $this->loader = $loader;
+        $this->logger = $logger;
+        $this->defaultLocale = $defaultLocale;
+        $this->endpoint = $endpoint;
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('%s://%s', LocoProviderFactory::SCHEME, $this->endpoint);
+    }
+
+    public function getName(): string
+    {
+        return LocoProviderFactory::SCHEME;
+    }
+
+    public function write(TranslatorBagInterface $translatorBag): void
+    {
+        $catalogue = $translatorBag->getCatalogue($this->defaultLocale);
+
+        if (!$catalogue) {
+            $catalogue = $translatorBag->getCatalogues()[0];
+        }
+
+        // Create keys on Loco
+        foreach ($catalogue->all() as $domain => $messages) {
+            $ids = [];
+            foreach ($messages as $id => $message) {
+                $ids[] = $id;
+                $this->createAsset($id);
+            }
+            if ($ids) {
+                $this->tagsAssets($ids, $domain);
+            }
+        }
+
+        // Push translations in all locales and tag them with domain
+        foreach ($translatorBag->getCatalogues() as $catalogue) {
+            $locale = $catalogue->getLocale();
+
+            if (!\in_array($locale, $this->getLocales())) {
+                $this->createLocale($locale);
+            }
+
+            foreach ($catalogue->all() as $messages) {
+                foreach ($messages as $id => $message) {
+                    $this->translateAsset($id, $message, $locale);
+                }
+            }
+        }
+    }
+
+    public function read(array $domains, array $locales): TranslatorBagInterface
```

Comment:
If I have two languages in Loco. `sv` and `en`. I add an asset, then add a translation to `en`. 

I run `translation:pull`, that will update `messages.en.xlf` and `messages.sv.xlf`. The swedish one is empty which prevents the fallback locale to act. 

Maybe dont download "untranslated"?

Reply:
Yes, it is possible using the `status` query parameter of the export API endpoint: https://localise.biz/api/docs/export/exportlocale